src/eos-updater-apply.c

```diff
@@ -103,6 +531,39 @@ apply_internal (EosUpdater *updater,
   if (!ostree_sysroot_load (sysroot, cancel, error))
     return FALSE;
 
+  flatpaks_this_commit_wants = flatpaks_to_deploy_from_commit (repo,
+                                                               update_id,
+                                                               cancel,
+                                                               &local_error);
+
+  /* If we don't have flatpaks to deploy, check the local_error
+   * to see what went wrong. If it was in the EOS_UPDATER_ERROR domain
+   * then something is wrong enough to warrant blocking the entire system
+   * update, otherwise it was probably just because we don't have the file
+   * in place */
+  if (!flatpaks_this_commit_wants &&
+      local_error &&
+      local_error->domain == EOS_UPDATER_ERROR)
+    {
+      g_propagate_error (error, local_error);
+      local_error = NULL;
```

Comment:
Nitpick: This can be simplified to `g_propagate_error (error, g_steal_pointer (&local_error))`, which has the bonus of making the ownership transfer even more explicit!